### ISSUE TITLE
isCollectionAccessible not using filters

### ISSUE THREAD

**Issue by RWOverdijk** [benwinding/react-admin-firebase#107]
When [fetching a single document](https://github.com/benwinding/react-admin-firebase/blob/master/src/providers/database/ResourceManager.ts#L83) you still use a [limit(1)](https://github.com/benwinding/react-admin-firebase/blob/master/src/providers/database/ResourceManager.ts#L171) to check if a collection "is accessible". Which on its own is already not the best approach to solve this, but is especially wrong when a single document is being fetched. The "permission check" is happening on an entirely arbitrary document which causes the whole thing to break down if there are document-level permissions (which is very common).

My suggestion is as follows: just fetch it, and if it fails check the error. No need for an extra query, ever. Especially if you're not even fetching the _same_ document to perform the check, causing permission errors that are really difficult to track down.

So basically stop the isCollectionAccessible mechanism altogether. Just remove the whole thing. The query will fail without it, too.

~~If you want I can do that and make a PR.~~ I did it locally already and it works perfectly (even when I don't have permission to do something).

#106

**Comment by benwinding** [comment]
Good points @RWOverdijk,
That was an older way, seems a bit ridiculous now. Appreciate the contribution back.
Cheers,
Ben
